packages/engineer/src/cli.ts

```diff
@@ -0,0 +1,116 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+
+/**
+ * We use Node's native module system to directly load configuration file.
+ * This configuration can (and should) be written as a `.ts` file.
+ */
+
+import program from 'commander';
+import { resolve } from 'path';
+import open from 'open';
+
+import { version } from '../package.json';
+import { resolvePackages, loadFeaturesFromPackages, runNodeEnvironment } from '@wixc3/engine-scripts';
+import { BaseHost } from '@wixc3/engine-core';
+import fs from '@file-services/node';
+import devServerFeature, { devServerEnv } from 'packages/engineer/feature/dev-server.feature';
+
+program.version(version);
+
+const parseBoolean = (value: string) => value === 'true';
+const collectMultiple = (val: string, prev: string[]) => [...prev, val];
+const defaultPublicPath = process.env.ENGINE_PUBLIC_PATH || '/';
+
+program
+    .command('start [path]')
+    .option('-r, --require <path>', 'path to require before anything else', collectMultiple, [])
+    .option('-f, --feature <feature>')
+    .option('-c, --config <config>')
+    .option('--mode <production|development>', 'mode passed to webpack', 'development')
+    .option('--inspect')
+    .option('-p ,--port <port>')
+    .option('--singleRun', 'when enabled, webpack will not watch files', false)
+    .option('--singleFeature', 'build only the feature set by --feature', false)
+    .option('--publicPath <path>', 'public path prefix to use as base', defaultPublicPath)
+    .option('--open <open>')
+    .option(
+        '--autoLaunch [autoLaunch]',
+        'should auto launch node environments if feature name is provided',
+        parseBoolean,
+        true
+    )
+    .option('--title <title>', 'application title to display in browser')
+    .option('--publicConfigsRoute <publicConfigsRoute>', 'public route for configurations')
+    .option('--engineerMode <gui|build>', 'interactive mode for engineer', 'gui')
+    .allowUnknownOption(true)
+    .action(async (path = process.cwd(), cmd: Record<string, any>) => {
+        const {
+            feature: featureName,
+            config: configName,
+            port: httpServerPort = 3000,
+            singleRun,
+            singleFeature,
+            open: openBrowser = 'true',
+            require: pathsToRequire,
+            publicPath = defaultPublicPath,
+            mode,
+            title,
+            publicConfigsRoute = 'configs/',
+            autoLaunch,
+            engineerMode,
+        } = cmd;
+        try {
+            const basePath = resolve(__dirname, '../../engineer');
+            preRequire(pathsToRequire, basePath);
+
+            const features = loadFeaturesFromPackages(resolvePackages(basePath), fs).features;
+
+            await runNodeEnvironment({
+                featureName: `engineer/${engineerMode as string}`,
```

Comment:
maybe allow for full feature name instead of engineer prefix?

Reply:
sure, to be able to expand it. 
as we discussed - If I will want to create a feature that extends the engineer/gui I should be able to, and I should be able to run it using this cli.
It should be, on the next phase, extendable

Reply:
Yeah I agree, changed it into engineerFeature, and the default is engineer/gui, whatever feature you provide would be the entry feature, although there is still the resolution issue, but one step at a time